### ISSUE TITLE
Grad norm cut in half every 2000 steps?

### ISSUE THREAD

**Issue by PCerles** [NVIDIA/apex#774]
Hi, I am quite confused over the purpose of the following line. 

https://github.com/NVIDIA/apex/blob/b9e5d37dbdbfbf5596acce3b915acf4b1a93a789/apex/amp/scaler.py#L214

My current model training does not have gradients overflow, but this line was cutting my gradient norm in half every 2000 steps up until the max grad scaler. Given that the default grad scaling starts at `2**16` and the default max grad scale is `2**24`, this line is shrinking the model gradient by half every 2000 steps up until 16000 steps, without having anything to do with dynamic loss scaling. This changes the convergence rate of the model (and was very mysterious to debug).

This is unexpected behavior and is affecting my training. I set the two to be the same now, but this feels like weird default behavior. What's the use of having the max grad scaler different from the initial grad scaler?

**Comment by daturkel** [comment]
@PCerles I know it's been quite a while since you posted this issue, but I was seeing some unusual gradients that were increasing in magnitude every 2k steps (regardless of batch size). I experimented with many settings and the only one which stopped this from happening was turning off mixed precision training. Can you help me understand what is happening here? I don't see any related impact to my loss function at those 2k intervals, despite the increases in the reported gradients. Is what I'm seeing just a bug in how the gradients are being reported to WandB?


<img width="1050" alt="Screen Shot 2023-10-23 at 12 44 24 PM" src="https://github.com/NVIDIA/apex/assets/1828387/4893c204-8527-4a63-99cf-a6ac93395f7d">
